sigilrecorder.lic

```diff
@@ -0,0 +1,290 @@
+
+		DRC.message("Target room (#{roomid}) matches in data = #{room_matches}")
+		
+		#return the number of instances of roomid
+		return room_matches
+	end #check_room_data
+	
+	#attempts to navigate out of a room lacking important data
+	def unstuck_me(target_room)
+		#Check the exits to the room we think we're in in Mapdb. Store the exits in an array then loop through it.
+		exit_list = Map[target_room].wayto
+		exit_list.each do |rnum, dir|
+		  #flag to listen for room numbers appearing in 'peer' info
+		  Flags.add('sigil-room-detected', /Room Number: \d+/)
+		  DRC.bput("peer #{dir}", /You peer/, /You see/, /\[Usage:/)
+		  pause 0.5
+		  #if we detected a room nubmer, move in that direction and return
+		  if Flags['sigil-room-detected']
+		    DRC.retreat
+		    fput(way)
+			Flags.reset('sigil-room-detected')
+			return
+		  end
+		end
+		#If we are unable to find a mapped exit, write the original target room to nogo and disconnect as a safety measure
+		DRC.message('Cannot find a way out. Exiting script.')
+		add_to_nogo(target_room, @data)
+		exit
+	end #unstuck_me
+	
+	#uses 'perc sigil' repeatedly until both sigils are discovered
+	def find_sigils
+		room_sigils = []
+		while room_sigils.length < 2
+			#weave is a 0 RT way of detecting enemies in the room. It will auto face enemies but will not harass players or NPCs. After weaving check if the danger-room flag has tripped
+			DRC.bput('weave', /You weave back and forth in a dance-like motion/, /You must be closer to use tactical abilities on your opponent/, /There is nothing else to face/)
+			if Flags['sigil-danger-room']
+			  add_to_nogo(Room.current.id, @data) unless Room.current.id == nil
+			  return nil
+			end 
+
+			#main 'perc sigil' DRC.bput and case statement. When sigils are detected they're added to an array
+			case line = DRC.bput('perceive sigil', @sigil_list, 'Roundtime', /^You recall.*/, /^Having recently been searched/, /You are too distracted/, /Something about|This is not an appropriate place for that/)
+			when /^You recall.*/
+			  DRC.message('Room searched too recently')
+			  return nil
+			when /You are too distracted/
+			  add_to_nogo(Room.current.id, @data)
+			  return nil
+			when /Something about|This is not an appropriate place for that/
+			  add_to_nogo(Room.current.id, @data)
+			  return nil
+			when /^Having recently been searched/
+			  return nil
+			when *@sigil_list
+			  room_sigils.push(line)
+			  DRC.message("Found: #{room_sigils}")
+			end
+		end #while
+
+		#return the array of all sigils discovered in the room (should be 2 unless something went wrong)
+		return room_sigils
+	end
+	
+	#generate the list of all rooms that contain the current room's Room.current.image data
+	def region_room_list
+		data = Map.list.select do |room|
+		  room.image == Room.current.image
+		end
+
+		#generate the array
+		roomlist = []
+		data.each do |room|
+		  roomlist.push(room.id)
+		end
+
+		#return the full list of rooms
+		return roomlist
```

Comment:
This loop is fine and how we have to do it most programming languages. I want to share another way it can be done in Ruby just for knowledge share.
```ruby
Map.list.select { |room| room.image == Room.current.image }.map(&:id)
```
**Output**
```
,e echo Map.list.select { |room| room.image == Room.current.image }.map(&:id)

--- Lich: exec1 active.

[exec1: [1746, 1747, 1748, 1749, 1750, 1751, 1752, 1753, 1754, 1755, 1756, 1757, 1758, 1759, 1760, 1761, 1762, 1763, 1764, 1765, 1766, 1767, 1768, 1769, 8439, 8440, 8441, 8442, 8443, 8444, 8445, 8446, 8447, 8448, 8449, 8450, 8451, 8452, 8453, 8454, 8455, 8456, 8457, 8458, 8459, 8460, 8461, 8462, 8463, 8464, 8465, 8466, 8467, 8468, 8469, 8470, 8471, 8472, 8473, 8474, 8475, 8476, 8477, 8478, 8479, 8480, 8481, 8482, 8483, 8484, 8485, 8486, 8487, 8488, 8489, 8490, 8491, 8492, 8493, 8494, 14105, 14106, 14107, 14108, 14109, 14110, 14111, 14112, 14113, 14114]]

--- Lich: exec1 has exited.
```

Reply:
Sorcery! Hope you didn't do that in town.

Reply:
Added the sorcery to my script. I'll need to do some Googling to truly understand it 🤣